gui/src/app/SPAnalysis/SPAnalysisReducer.ts

```diff
@@ -23,13 +23,20 @@ export type SPAnalysisReducerAction = {
     type: 'setSamplingOpts',
     opts: Partial<SamplingOpts>
 } | {
-    type: 'loadLocalStorage',
+    type: 'loadInitialData',
     state: SPAnalysisDataModel
 } | {
     type: 'clear'
 }
 
-export const SPAnalysisReducer: SPAnalysisReducerType = (s: SPAnalysisDataModel, a: SPAnalysisReducerAction) => {
+export const SPAnalysisReducer = (onDirty: () => void) => (s: SPAnalysisDataModel, a: SPAnalysisReducerAction) => {
+    if (a.type !== "loadInitialData") {
+        // TextEditor seems to trigger occasional spurious edits where nothing changes
+        if (a.type !== "editFile" || s[a.filename] != a.content) {
```

Comment:
> Or maybe the desired behavior is just "clear the URL bar if the user breathes on anything" and we don't need to worry about this.

This is the behavior I would argue for. Even editing the sampler parameters when those weren't set in the url does break the rule of "is this the page I would view if I clicked on the link again"

Reply:
Well, at that point it's kind of just indeterminate--if there's any part of the project that isn't set, you can't guarantee that you'll get the unset parts correct when you load, because they just aren't there. (This is more of a philosophical point right now, I'm not advocating for any particular practical decision as a result of this observation)

Reply:
It would be the default value still, which is well-defined even if it isn't necessarily something we're guaranteeing will never change.

I agree something here could be more sophisticated, but for now it not being so is intentional

Reply:
Ahhh, okay. Confirmed that it does use the default values, not the last values.